storage/range.go

```diff
@@ -208,7 +208,17 @@ func NewRange(desc *proto.RangeDescriptor, rm RangeManager) (*Range, error) {
 	}
 	r.SetDesc(desc)
 
-	err := r.loadLastIndex()
+	var zoneConfig *proto.ZoneConfig
```

Comment:
This value is set by the store very efficiently for all of the ranges in setRangeMaxBytes. Doing it here would be quite inefficient (keep in mind that there may be a million ranges in a single store, so this means every range at startup would have to call through to read the zone config).

Also, you're only getting the default zone config here, which won't apply to all ranges. Further, reading the zone config directly from the engine only will work if the store contains the range with that zone config. And one more thing--reading this with proto.ZeroTimestamp won't work as zone configs are stored using MVCC. You'd need to use the current time or else you'd end up missing the written value as the timestamp would be earlier than any written value.